logic.js

```diff
@@ -302,23 +302,44 @@ function create_input(type, name, value, text)
 function apply_deck_selection(decks)
 {
     var container = document.getElementById("tableau");
-    container.innerHTML = ""; // TODO use deck.discard_deck instead
+
+    select(
+        visible_decks,
+        function(visible_deck) {
+            return !any(decks, function(d) { return d.name === visible_deck.name; });
+        }
+    ).forEach(function(deck) { deck.discard_deck(); });
 
     for (var i = 0; i < decks.length; i++)
     {
+        var deck = decks[i];
+
+        if (any(visible_decks, function(visible_deck) { return visible_deck.name === deck.name; })) {
+            // This deck is already on the tableau.
+            continue;

```

Comment:
I approve of style nitpicks. 👍 Revised to use `.forEach` loop on the filtered array.